### PR TITLE
Fix the error of incremental tasks caused by datetime containing zero value in MySQL

### PR DESCRIPTION
**Pull request by azexcy**
refer https://dev.mysql.com/doc/refman/8.0/en/datetime.html
MySQL may have the datetime value wiht '0000-00-00 00:00:00', but the value is illegal for `LocalDateTime` at java.

Changes proposed in this pull request:
  - Fix the error of incremental tasks caused by datetime containing zero value in MySQL
  - Fix MySQLClient always print error log when decode binlog failed.

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Review by sandynz — CHANGES_REQUESTED** [review]

File: db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLDatetime2BinlogProtocolValue.java [hunk 1]
```diff
@@ -48,6 +48,9 @@ private long readDatetimeV2FromPayload(final MySQLPacketPayload payload) {
     
     private Serializable readDatetime(final MySQLBinlogColumnDef columnDef, final long datetime, final MySQLPacketPayload payload) {
         long datetimeWithoutSign = datetime & (0x8000000000L - 1);
+        if (0 == datetimeWithoutSign) {
+            return MySQLTimeValueUtils.DATETIME_OF_ZERO;
+        }
```
[the remark this answers is not in the document]
Comment (sandynz) [reply db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLDatetime2BinlogProtocolValue.java:53]: https://dev.mysql.com/doc/internals/en/date-and-time-data-type-representation.html is not available now, refer to https://stackoverflow.com/questions/36937211/how-datetime-values-are-stored-in-mysql for more details.

[the remark this answers is not in the document]
Comment (sandynz) [reply db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLDatetime2BinlogProtocolValue.java:53]: The sign bit might be needed

[the remark this answers is not in the document]
Comment (azexcy) [reply db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLDatetime2BinlogProtocolValue.java:53]: The official document not find currently, but i found some relate source code from https://github.com/mysql/mysql-server/blob/5.7/sql-common/my_time.c#L1783

image

the first bit sign seem not related to time

**Comment by azexcy** [comment]
I find a problem when init ShardingSphereDataSource, the parameter at `jdbcUrl` maybe not equals with `dataSourceProperties`

image

This may lead to results that do not match expectations

**Review by sandynz — APPROVED** [review]

**Pull request merged**